code/unclassified/src/odd-even-sort/odd_even_sort.c

```diff
@@ -0,0 +1,55 @@
+#include <stdio.h>
+#include<math.h>
+#include "mpi.h"
+
+void swap(int array[], int i, int j)
+{
+        int temp = array[i];
+        array[i] = array[j];
+        array[j] = temp;
+}
+
+int main(int argc, char** argv)
+{
+        int my_rank, p, res, dest = 0, tag = 0, i;
+
+        int array[8] = {9, 2, 1, 3, 5, 4, 6, 7};
+        int n = 8;
+
+        MPI_Status status;
```

Comment:
Why do you even need MPI anyway?

Reply:
For message passing in the parallelized structures.